nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -128,6 +128,16 @@ in
         '';
       };
 
+      firewallOpenPorts = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Specifies whether the ports the SSH daemon listens on must
+          be enabled in the firewall. Remember to open at least some
+          of them.
```

Comment:
> Could you be more specific as to why it might be useful to combine `services.openssh.listenAddresses` with this PR?

We don't necessarily have to combine it with this PR, but Imho it is a good idea to consider this since you made related modifications inside the openssh module (and at least we have to get the description right!).

> Imust confess I don't really understand how to use `services.openssh.listenAddresses`

It doesn't modify the firewall (atm) so you have to do that manually (I thought this is how you've locked yourself out in the first place).

By default OpenSSH will listen on port `22` on all IP addresses of all interfaces, with this option you could e.g. limit that to `[::1]:22` (yes, that example doesn't make much sense...).

> it sends RST packets (revealing that there's an SSH server running on the machine)

I assume that RST packet (TCP reset) came from the firewall. Then you should only get those for existing connections (I guess that terminated your running SSH session).

> "iptables -F" to re-enable everything, whereas `listenAddresses`

The change is intended to prevent a user from getting into that situation (not to fix a "broken" firewall) :wink: